src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1485,8 +1485,8 @@ public static MobEffect loadMobEffect(CompoundTag nbt, String key, @Nullable Mob
         }
     }
   
-    private static BannerPattern[] nonPatternItems;
-    private static int totalPatternRows;
+    private static BannerPattern[] nonPatternItems= Arrays.stream(BannerPattern.values()).filter(p -> !p.hasPatternItem).toArray(BannerPattern[]::new);
+    private static int totalPatternRows = (nonPatternItems.length + 2) / 4;
```

Comment:
Perhaps the `refreshBannerPatternData` method could be called from a static initializer immediately below this method, to avoid duplicating code between the field initializers and the methods?